app/controllers/api/v2/discovered_hosts_controller.rb

```diff
@@ -80,7 +80,7 @@ def create
       end
 
       def update
-        @host = ::ForemanDiscovery::HostConverter.to_managed(@discovered_host, true, true, managed_host_params)
+        @host = ::ForemanDiscovery::HostConverter.to_managed(@discovered_host, true, managed_host_params[:build], managed_host_params)
```

Comment:
Introduced in 044cd42ffcb96819af0bd3d88bb348f2f9dc8ea9 and regression - users were unable to override build flag to false, it was always true.